python/lsst/ip/diffim/getTemplate.py

```diff
@@ -474,3 +479,219 @@ def runDataRef(self, *args, **kwargs):
+        coaddExposureList = []
+        for coaddRef in inputs['coaddExposures']:
+            dataId = coaddRef.dataId
+            patchWcs = inputs['skyMap'][dataId['tract']].getWcs()
+            patchBBox = inputs['skyMap'][dataId['tract']][dataId['patch']].getOuterBBox()
+            patchCorners = patchWcs.pixelToSky(geom.Box2D(patchBBox).getCorners())
+            patchPolygon = afwGeom.Polygon(inputs['wcs'].skyToPixel(patchCorners))
+            if patchPolygon.intersection(detectorPolygon):
+                overlappingArea += patchPolygon.intersectionSingle(detectorPolygon).calculateArea()
+                self.log.info("Using template input tract=%s, patch=%s" %
+                              (dataId['tract'], dataId['patch']))
+                coaddExposureList.append(coaddRef)
+
+        if not overlappingArea:
+            raise pipeBase.NoWorkFound('No patches overlap detector')
+
+        return coaddExposureList
+
+    def run(self, coaddExposures, bbox, wcs):
+        """Warp coadds from multiple tracts to form a template for image diff.
+
+        Where the tracts overlap, the resulting template image is averaged.
+        The PSF on the template is created by combining the CoaddPsf on each
+        template image into a meta-CoaddPsf.
+
+        Parameters
+        ----------
+        coaddExposures: list of DeferredDatasetHandle to `lsst.afw.image.Exposure`
+            Coadds to be mosaicked
+        bbox : `lsst.geom.Box2I`
+            Template Bounding box of the detector geometry onto which to
+            resample the coaddExposures
+        wcs : `lsst.afw.geom.SkyWcs`
+            Template WCS onto which to resample the coaddExposures
+
+        Returns
+        -------
+        result : `struct`
+            return a pipeBase.Struct:
+            - ``outputExposure`` : a template coadd exposure assembled out of patches
+
+        """
+        # Table for CoaddPSF
+        tractsSchema = afwTable.ExposureTable.makeMinimalSchema()
+        tractKey = tractsSchema.addField('tract', type=np.int32, doc='Which tract')
+        patchKey = tractsSchema.addField('patch', type=np.int32, doc='Which patch')
+        weightKey = tractsSchema.addField('weight', type=float, doc='Weight for each tract, should be 1')
+        tractsCatalog = afwTable.ExposureCatalog(tractsSchema)
+
+        finalWcs = wcs
+        bbox.grow(self.config.templateBorderSize)
+        finalBBox = bbox
+
+        nPatchesFound = 0
+        maskedImageList = []
+        weightList = []
+
+        for coaddExposure in coaddExposures:
+            coaddPatch = coaddExposure.get()
+
+            # warp to detector WCS
+            xyTransform = afwGeom.makeWcsPairTransform(coaddPatch.getWcs(), finalWcs)
+            psfWarped = WarpedPsf(coaddPatch.getPsf(), xyTransform)
+            warped = self.warper.warpExposure(finalWcs, coaddPatch, maxBBox=finalBBox)
+
+            # Check if warped image is viable
+            if warped.getBBox().getArea() == 0:
```

Comment:
When working with what I believe is very similar code, this check didn't trigger when there wasn't overlap, but adding `or not np.any(np.isfinite(warped.getImage().array))` worked to check that it wasn't a viable warped image and should be skipped.